test/acceptance/kafo_configure_test.rb

```diff
@@ -173,5 +173,48 @@ module Kafo
         _(File.exist?("#{INSTALLER_HOME}/testing")).must_equal true
       end
     end
+
+    describe 'the configuration file' do
+      before do
+        @saved_values = Kafo::Configuration::DEFAULT.reject do |key, value|
+          value.nil?
+        end
+      end
+
+      it 'must only save defined values' do
+        code, _, err = run_command '../bin/kafo-configure'
+
+        _(code).must_equal 0, err
+        _(YAML.load_file(KAFO_CONFIG).keys.sort).must_equal @saved_values.keys.sort
+      end
+
+      it 'must not save non-persisted command line options' do
+        config = YAML.load_file(KAFO_CONFIG)
+        config[:colors] = true
+        File.open(KAFO_CONFIG, 'w') do |file|
+          file.write(config.to_yaml)
+        end
+
+        code, _, err = run_command '../bin/kafo-configure --noop --no-colors'
+
+        _(code).must_equal 0, err
+        _(YAML.load_file(KAFO_CONFIG).keys.sort).must_equal @saved_values.keys.sort
+        _(YAML.load_file(KAFO_CONFIG).keys).wont_include 'noop'
+        _(YAML.load_file(KAFO_CONFIG)[:colors]).must_equal false
```

Comment:
This is precisely the kind of test I meant. As a user you might expect that `--noop` doesn't save anything, but here it does. That surprises me.

I do wonder if the autodetection that colors has makes a difference, but using `color_of_background` doesn't have that and can certainly answer the question.

Reply:
Ohh, now I see what you were asking for. I will update the test (note that I am passing --colors so this is exactly as expected).

Reply:
Or are you just pointing to the expectations of `noop` itself? I do not think it works as you are expecting today either and I'm not sure it should. I grant there are multiple levels on which to `interpret` noop, but it's a bit trickier with the options destined for the config file. From testing, if you pass `--noop --ignore-undocumented` today with kafo, the value gets saved to config file. I think this would have to be a separate bug fix.

Reply:
So after taking some consideration from https://github.com/theforeman/kafo/pull/291 and rebasing this on latest, I ended up with a solution that was needed to solve the target of this issue and it in turn addresses your concern here over noop. The tests have been updated to reflect this.